internal/controllers/machine/machine_controller_phases.go

```diff
@@ -312,3 +312,44 @@ func (r *Reconciler) reconcileInfrastructure(ctx context.Context, cluster *clust
 	m.Spec.ProviderID = pointer.StringPtr(providerID)
 	return ctrl.Result{}, nil
 }
+
+func (r *Reconciler) reconcileCertificateExpiry(ctx context.Context, _ *clusterv1.Cluster, m *clusterv1.Machine) (ctrl.Result, error) {
+	var annotations map[string]string
+
+	if !util.IsControlPlaneMachine(m) {
+		// If the machine is not a control plane machine, return early.
+		return ctrl.Result{}, nil
+	}
+
+	if m.Spec.Bootstrap.ConfigRef != nil {
+		bootstrapConfig, err := external.Get(ctx, r.Client, m.Spec.Bootstrap.ConfigRef, m.Namespace)
+		if err != nil {
+			return ctrl.Result{}, err
```

Comment:
I think we have to wrap the error here. Otherwise it could become hard to debug

I would prefix/wrap all err's we return in this func with "failed to reconcile certificate expiry" (including the parse errors below)